packages/wallet-manager/src/wallets/Evm/Evm.ts

```diff
@@ -0,0 +1,200 @@
+import { ethers } from 'ethers';
+import events from 'events';
+import type {
+  Web3Provider,
+  ExternalProvider,
+  Provider
+} from '@ethersproject/providers';
+import { IEvmWallet } from '../../interfaces';
+import { AddChain } from '../../types';
+
+class EvmWallet extends events.EventEmitter implements IEvmWallet {
+  address?: string;
+  signer?: ethers.Signer;
+  web3Provider!: Web3Provider;
+  windowConnector: ExternalProvider;
```

Comment:
Not sure we actually need to keep this as a property. The underlying provider is accessible via the `web3Provider.provider` property.

Reply:
Let me check on this and I'm going to remove the attribute if necessary

Reply:
Checking this, the thing is that, if you don't pass provider to the class, it uses `window.ethereum`. Then, web3Provider, being initialized from it, get assigned if the condition is met. Since `web3Provider` doesn't overlap in types with `window.ethereum`, I maintain those separate to listen to the events. What would be your suggestion to remove this property?

Reply:
Basically, I would just expand the type when adding the listeners, but not keep the `windowConnector` on state, as it is already accessible via `this.web3Provider.provider`.

In all places where `this.windowConnector as Provider` is called should be replaced with `this.web3Provider.provider as Provider` and everything should work the same.

Reply:
done